### PR TITLE
fix: reject prefixItems in strict tool schemas

### LINKED ISSUE #4752: Strict schemas still emit unsupported prefixItems for ordinary fixed-tuple parameters
### What happened?

A fixed-length tuple on an ordinary function parameter still produces `prefixItems` in a strict tool schema:

```python
from agents.function_schema import function_schema

def locate(point: tuple[int, str]) -> str:
    return repr(point)

schema = function_schema(locate, strict_json_schema=True)
print(schema.params_json_schema[properties][point])
```

On current `main` (`89c02c8`), this prints:

```python
{
    maxItems: 2,
    minItems: 2,
    prefixItems: [{type: integer}, {type: string}],
    title: Point,
    type: array,
}
```

Tool construction succeeds, so the unsupported schema is deferred to the provider request.

### Why this is a bug

In #4696, the maintainer decision was explicit:

> emitting `prefixItems` is not a supported direction for OpenAI Structured Outputs

and:

> fixed-length tuple annotations [should be rejected] during tool construction and identify `tuple[T, ...]` or `list[T]` as the supported alternative

#4735 implements that fail-fast behavior for `*args: tuple[int, str]`, but the ordinary-parameter path still passes the same unsupported schema through `function_schema()`.

This is not the `*args` widening bug fixed by #4735. It is the remaining public construction path that emits the provider-invalid keyword under strict mode.

### Expected behavior

When `strict_json_schema=True`, constructing a function tool with an ordinary fixed-length tuple parameter should fail early with an actionable `UserError` that names `tuple[T, ...]` and `list[T]` as supported alternatives.

When `strict_json_schema=False`, the existing fixed-tuple Pydantic schema and validation should remain available for custom/non-OpenAI providers that accept `prefixItems`.

### Scope

The same construction-time guard should be checked for structured output types as well, because Pydantic emits `prefixItems` there too. The narrow invariant is: strict schemas must not leave SDK construction containing a keyword that the supported provider subset rejects.

### Environment

- `openai-agents-python` current `main`: `89c02c828ee8510fe9a84ee6675608193aa13b02`
- Python 3.12.13
- Pydantic 2.12.3
- Reproduced offline; no API request or credentials required.

### PR DESCRIPTION
**Pull request by mikemikimike**
### Summary

Reject `prefixItems` during strict JSON schema conversion so ordinary fixed-length tuple parameters fail during tool construction with guidance for supported homogeneous tuples and lists. Non-strict schemas retain their existing behavior.

### Test plan

- `uv run --frozen pytest tests/test_function_schema.py tests/test_strict_schema.py tests/test_strict_schema_oneof.py -q` (130 passed)
- `uv run --frozen ruff check src/agents/strict_schema.py tests/test_function_schema.py` (passed)
- `uv run --frozen ruff format --check src/agents/strict_schema.py tests/test_function_schema.py` (passed)
- `uv run --frozen pyright src/agents/strict_schema.py` (0 errors)
- `git diff --check` (passed)

The full pytest suite and single-file mypy check did not complete in the local Windows environment; both remained in dependency/test execution without producing a result and were terminated. No production failure was observed.

### Issue number

Closes #4752

### Checks

- [x] I've added new tests, if relevant
- [ ] I've run `.agents/skills/code-change-verification/scripts/run.sh`
- [ ] I've confirmed all verification steps pass
- [x] If using Codex, I've run `/review` before submitting this PR

### CODE REVIEW

**Comment by XuQuanxin04** [comment]
Automated Codex review identified a compatibility edge case in the current approach: traversing every dict while excluding annotation keys can reject a valid object property literally named `prefixItems` (and can skip schemas beneath property names such as `default`/`examples`). I reproduced this with a strict schema containing `properties: {"prefixItems": {"type": "string"}}`.

The traversal should follow schema-valued positions and treat entries of `properties`, `$defs`, `definitions`, `patternProperties`, and `dependentSchemas` as subschemas regardless of their names. I prepared a focused regression test and corrected walker locally; commit `96cbb4a` in my fork branch (push retry may be needed due transient network).

**Comment by mikemikimike** [comment]
Addressed the compatibility edge case from the review comment in commit `54b6e83c`.

The strict-schema rejection now traverses only schema-valued JSON Schema positions, including `properties`, `$defs`, `definitions`, `patternProperties`, `dependentSchemas`, unions, and conditional keywords. Literal property names such as `prefixItems`, and annotation values under `default`/`examples`, are preserved without false positives, while an actual `prefixItems` schema is still rejected during construction.

Validation:
- `uv run --frozen pytest tests/test_function_schema.py tests/test_strict_schema.py tests/test_strict_schema_oneof.py -q` — 134 passed
- `ruff format --check src/agents/strict_schema.py tests/test_strict_schema.py` — passed
- `ruff check src/agents/strict_schema.py tests/test_strict_schema.py` — passed
- `uv run --frozen pyright src/agents/strict_schema.py tests/test_strict_schema.py` — 0 errors
- `git diff --check` — passed

The repository-level `make typecheck` is not runnable as written on Windows because its parallel shell recipe uses POSIX syntax; direct Pyright passed, while repository mypy remains blocked by the pre-existing missing `numpy` stubs in `src/agents/voice/imports.py`. The full test command was attempted with the Windows-compatible equivalent, but xdist hit a permissions error in the system pytest temp directory; the single-process focused and three-file suites pass.

File: src/agents/strict_schema.py [hunk 1]
```diff
@@ -36,6 +36,60 @@
...
+)
+_SCHEMA_CHILD_LIST_KEYS = frozenset({"allOf", "anyOf", "oneOf"})
+_SCHEMA_CHILD_MAP_KEYS = frozenset(
+    {"$defs", "definitions", "dependentSchemas", "patternProperties", "properties"}
+)
+
+
+def _reject_prefix_items(schema: object) -> None:
+    """Reject ``prefixItems`` in schema-valued positions without inspecting annotations."""
+    stack = [schema]
```
[the remark this answers is not in the document]
Comment (mikemikimike) [reply src/agents/strict_schema.py]: Addressed in commit `359c446b`: removed the standalone traversal tables and helper, and kept the `prefixItems` check inside `_ensure_strict_json_schema()`, which already follows the schema nodes emitted by Pydantic. The focused regression suite remains green: 130 passed.

**Comment by seratch** [comment]
Thanks for working on this. After reconsidering the scope, I don't think we should add a blanket `prefixItems` rejection to the shared strict-schema converter.

#4735 addresses a separate problem with variadic parameters. For ordinary fixed-length tuple parameters, the provider already rejects unsupported schemas before the tool runs. This change moves that rejection earlier, but we have not identified a concrete problem that warrants maintaining an additional SDK-side restriction.

For these inputs, please use a supported homogeneous collection such as `list[T]` or `tuple[T, ...]`, or a structured object with named fields.

My earlier guidance in #4696 encouraged this broader validation, and I should have been more precise. Sorry for the unnecessary work that caused. I'm closing this as not planned.
